wasm-text.md

```diff
@@ -744,28 +775,34 @@ They distribute the text-to-abstract functions above over lists.
 
     rule #t2aInstrs<C>(I:Instr IS:Instrs) => #t2aInstr<C>(I) #t2aInstrs<C>(IS)
     rule #t2aInstrs<_>(.Instrs) => .Instrs
-
-    rule #t2aLocalDecls<C>(LD:LocalDecl LDS:LocalDecls) => #t2aLocalDecl<C>(LD) #t2aLocalDecls<C>(LDS)
-    rule #t2aLocalDecls<_>(.LocalDecls) => .LocalDecls
 ```
 
 ### Functions for Gathering Context
 
 The following are helper functions for gathering and updating context.
 
 ```k
-    syntax Map ::= #idcFuncs    ( Defns, Defns      ) [function]
-                 | #idcFuncsAux ( Defns, Defns, Int ) [function]
- // ------------------------------------------------------------
-    rule #idcFuncs(IMPORTS, DEFNS) => #idcFuncsAux(IMPORTS, DEFNS, 0)
+    syntax Map ::= #idcTypes    ( Defns           ) [function]
+                 | #idcTypesAux ( Defns, Int, Map ) [function]
+ // ----------------------------------------------------------
+    rule #idcTypes(DEFNS) => #idcTypesAux(DEFNS, 0, .Map)
+
+    rule #idcTypesAux((type ID:Identifier (func _)) TS => TS, IDX => IDX +Int 1,  ACC => ACC [ ID <- IDX ]) requires notBool ID in_keys(ACC)
+    rule #idcTypesAux((type               (func _)) TS => TS, IDX => IDX +Int 1, _ACC)
+    rule #idcTypesAux(.Defns, _, ACC) => ACC
```

Comment:
So the plan is to just get stuck if the same ID is used twice?

Reply:
Seems reasonable, since it's an error in the input program.

Reply:
Yep.